client/scripts/views/components/new_thread_form.ts

```diff
@@ -159,16 +160,37 @@ export const NewThreadForm: m.Component<{
     vnode.state.form = {};
     vnode.state.recentlyDeletedDrafts = [];
     vnode.state.uploadsInProgress = 0;
-    if (vnode.state.newType === undefined) {
-      vnode.state.newType = localStorage.getItem(`${app.activeId()}-post-type`) || PostType.Discussion;
+    vnode.state.overwriteConfirmationModal = false;
+    if (vnode.state.postType === undefined) {
+      vnode.state.postType = localStorage.getItem(`${app.activeId()}-post-type`) || PostType.Discussion;
     }
-    if (vnode.state.newType === PostType.Discussion) {
+    if (vnode.state.postType === PostType.Discussion) {
       vnode.state.form.threadTitle = localStorage.getItem(`${app.activeId()}-new-discussion-storedTitle`);
     } else {
       vnode.state.form.url = localStorage.getItem(`${app.activeId()}-new-link-storedLink`);
       vnode.state.form.linkTitle = localStorage.getItem(`${app.activeId()}-new-link-storedTitle`);
     }
   },
+  onremove: async (vnode) => {
+    const { fromDraft, form, quillEditorState, postType, overwriteConfirmationModal } = vnode.state;
+    if (postType === PostType.Discussion && !overwriteConfirmationModal) {
+      if (quillEditorState?.alteredText) {
+        let confirmed = false;
+        const modalMsg = fromDraft
+          ? 'Update saved draft?'
+          : 'Save as draft?';
+        confirmed = await confirmationModalWithText(modalMsg)();
+        if (confirmed) {
+          await saveDraft(form, quillEditorState, null, fromDraft);
+          notifySuccess('Draft saved');
+        }
+      }
+      localStorage.removeItem(`${app.activeId()}-new-discussion-storedTitle`);
+      localStorage.removeItem(`${app.activeId()}-new-discussion-storedText`);
+      localStorage.removeItem(`${app.activeId()}-active-tag`);
+      localStorage.removeItem(`${app.activeId()}-post-type`);
```

Comment:
I feel like ideally the prompt should cancel the modal from closing if "Cancel" is pressed but that may be too difficul to do easily?

Reply:
I'll switch up the copy for clarity